extensions/blocks/mailchimp/view.js

```diff
@@ -41,22 +45,41 @@ function fetchSubscription( blogId, email, params ) {
 	} );
 }
 
+function validateEmail( form, emailField ) {
+	const email = emailField.value;
+	const errorClass = 'error';
+	emailField.classList.remove( errorClass );
+	if ( ! emailValidator.validate( email ) ) {
+		emailField.classList.add( errorClass );
+		if ( typeof document.createElement( 'input' ).reportValidity === 'function' ) {
+			// In case that browser supports it, trigger HTML5 validation
+			form.reportValidity();
+		}
+		return false;
+	}
+
+	return true;
+}
+
 function activateSubscription( block, blogId ) {
 	const form = block.querySelector( 'form' );
-	const errorClass = 'error';
+	const emailField = block.querySelector( 'input[name=email]' );
 	const processingEl = block.querySelector( '.' + blockClassName + '_processing' );
 	const errorEl = block.querySelector( '.' + blockClassName + '_error' );
 	const successEl = block.querySelector( '.' + blockClassName + '_success' );
+	emailField.addEventListener(
```

Comment:
Do we need to remove the event listener as well?

Reply:
It is an enhancement so that errors are reported as you type (and when you click submit). We can remove it if you think we should.

```
emailField.addEventListener(
		'input',
		debounce( () => {
			validateEmail( form, emailField );
		}, 1000 )
	);
```

Reply:
I didn't see a corresponding `removeEventListener` when skimming the code, that's why I was asking.

Reply:
Ah, now I got you! Although I can't see the case of the event being registered again, we should properly remove event listeners. Addressed here https://github.com/Automattic/jetpack/pull/16329/commits/39c4b225500e9e25933a9aa104bd9ab77939847d